File: sdkwork-im-sdk-typescript/composed/src/auth-module.ts
import type {
  ImAuthLoginRequest,
  ImAuthLoginResult,
  ImAuthSession,
} from './types.js';
import type { ImSdkContext } from './sdk-context.js';

export class ImAuthModule {
  constructor(private readonly context: ImSdkContext) {}

  async login(body: ImAuthLoginRequest): Promise<ImAuthLoginResult> {
    const session = await this.context.transportClient.auth.login(body);
    if (session.accessToken) {
      this.useToken(session.accessToken);
    }
    return session;
  }

  me(): Promise<ImAuthSession> {
    return this.context.transportClient.auth.me();
  }

  useToken(token: string): void {
    this.context.setAuthToken(token);
  }

  clearToken(): void {
    this.context.clearAuthToken();
  }
}

File: sdkwork-im-sdk-flutter/composed/lib/src/auth_module.dart
import 'package:im_sdk_generated/im_sdk_generated.dart';

import 'context.dart';

class ImAuthModule {
  final ImSdkContext context;

  ImAuthModule(this.context);

  Future<PortalLoginResponse?> login(PortalLoginRequest body) async {
    final session = await context.transportClient.auth.login(body);
    final accessToken = session?.accessToken;
    if (accessToken != null && accessToken.isNotEmpty) {
      useToken(accessToken);
    }
    return session;
  }

  Future<PortalMeResponse?> me() {
    return context.transportClient.auth.me();
  }

  void useToken(String token) {
    context.setAuthToken(token);
  }

  void clearToken() {
    context.clearAuthToken();
  }
}
